components/server/src/workspace/garbage-collector.ts

```diff
@@ -169,20 +169,33 @@ export class WorkspaceGarbageCollector {
             const workspaces = await this.workspaceDB
                 .trace({ span })
                 .findVolumeSnapshotWorkspacesForGC(this.config.workspaceGarbageCollection.chunkLimit);
-            const volumeSnapshots = await Promise.all(
-                workspaces.map((ws) =>
-                    this.workspaceDB
+            const volumeSnapshotsWithWSPromise = await Promise.all(
```

Comment:
```suggestion
            const volumeSnapshotsWithWS = await Promise.all(
```

Reply:
Ah ,wait, there_s something else going on here. :eyes:

Reply:
@sagor999 Instead of doing it in here I pushed a commit here to make this code a tad more idiomatic: https://github.com/gitpod-io/gitpod/tree/pavel/vsWithType-suggestion.